### PR TITLE
chore(security): document sanitization breaking change.

### PR DESCRIPTION
**Pull request by mprobst**
See #8491.

### CODE REVIEW

**Comment by ericmartinezr** [comment]
With all due respect this is simply stupid

`this.safeStyleValue = sanitizer.bypassSecurityTrustStyle('rotate(90deg)')` 

Angular2 should have an internal list of *dangerous* styles and pass them through that ugly function without the user knowing about, this way sanitization can even be optional `provide(Sanitization, {useValue: true/false`` (or whatever)

PLEASE don't make the dom handling the new router, PLEASE

**Comment by mprobst** [comment]
@ericmartinezr we should improve our CSS sanitization whitelist, stuff like `rotate(7deg)` should be reasonable to detect. We'll have an issue on that.

That being said, you cannot detect all dangerous patterns. There's always some odd browser or some crazy escaping hack that would break app security. See e.g. [OWASP's resources on XSS prevention](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet) for more documentation.

**Comment by ericmartinezr** [comment]
>you cannot detect all dangerous patterns

In that case angular2 shouldn't do anything about it, or you provide full security or you provide none. Let the user handle the security of his/her own application, and in that case (again), Sanitization optional would be a solution/workaround. If I know angular2 can't provide full security, I turn sanitization off and I do it myself, if I know it will provide little security (and I'm ok with that) I turn it on and I fill the gaps ng2 can't (use that ugly function above, or any other).

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+<a name="2.0.0-rc.2"></a>
+### OTHER BREAKING CHANGES
+
+- Style values and URLs are now automatically sanitized. Values that do not match are escaped or
+  ignored. When binding a URL or style property that would get ignored, bind to a value explicitly
+  marked as safe instead by injection the DOM sanitization service:
+      MyComponent(sanitizer: DomSanitizationService) {
+        // ONLY DO THIS FOR VALUES YOU KNOW TO BE SAFE! NEVER ALLOW USER DATA IN THIS!
```
Comment (zoechi) [inline CHANGELOG.md]: Shouldn't this be
```ts
MyComponent {
  constructor(sanitizer: DomSanitizationService) {
```

**Comment by CaptainCodeman** [comment]
Yes, please let this be opt-in or at least able to be turned off.

If your styles are 100% app code generated or you properly sanitize content then it's not required.

File: CHANGELOG.md (hunk 1 above)
Comment (mprobst) [inline CHANGELOG.md]: Good catch, I was trying to be brief and failed :-) done.

**Comment by mprobst** [comment]
I filed #8511 for a general tracking bug on this, let's move the discussion there (this PR is only about documenting it).

**Comment by mprobst** [comment]
@ericmartinezr the whitelist implementation is fully secure, by only allowing known to be safe values. I'm just explaining why we cannot have a blacklist based implementation as you suggested.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+<a name="2.0.0-rc.2"></a>
+### OTHER BREAKING CHANGES
+
+- Style values and URLs are now automatically sanitized. Values that do not match are escaped or
+  ignored. When binding a URL or style property that would get ignored, bind to a value explicitly
+  marked as safe instead by injection the DOM sanitization service:
+      class MyComponent {
```
Comment (IgorMinar) [inline CHANGELOG.md]: it's usually safer to use tripple back-tick style escaping of code blocks. using indentation is error prone and can get messed up with reformatting.

**Comment by IgorMinar** [comment]
lgtm

File: CHANGELOG.md (hunk 2 above)
Comment (mprobst) [inline CHANGELOG.md]: Done.

**Pull request closed without merging**